master/internal/api_auth.go

```diff
@@ -29,6 +30,10 @@ func replicateClientSideSaltAndHash(password string) string {
 
 func (a *apiServer) Login(
 	_ context.Context, req *apiv1.LoginRequest) (*apiv1.LoginResponse, error) {
+	if len(os.Getenv("EXTERNAL_JWT_KEY")) > 0 {
```

Comment:
I wasn't a big fan of doing this via environment variables, but the config isn't easily accessible from some of the packages we need to make this decision, like early in the process of handling protobuf requests.

Reply:
I feel pretty strongly this has to go in into the config. You can even wire up config variables to be settable by environment variables if it's necessary, but I don't want to have random env vars read all over the master code, I want the source of truth to be the master config (which is already confusing enough as is, tbh). 

> but the config isn't easily accessible from some of the packages we need to make this decision, like early in the process of handling protobuf requests.

What exactly do you mean by this?

Reply:
grpcutil is the tricky part where we're authenticating with very little context built up. Attempting to use master config from in there created a mess of circular dependencies. After chatting yesterday I tried a few times to refactor this. What I've ended up with works and avoids the use of environment variables. But it does mean the master config is split up further - moving it to pkg ends up being a massive refactor and eventually around the storage configs it starts creating even worse circular dependencies. So it's not as clean as I'd like, but it does mean we use the config, of pieces of it.

Reply:
yeah, i think it's fine for now.. i think i can one click (with goland) refactor the entire mess to like `pkg/master_config` and never think about it again.

Reply:
maybe i'll see what https://github.com/rsc/rf can do one day

Reply:
we do have a 'master config' cleanup on the books along the lines of the expconf stuff, just never enough bandwidth